### ISSUE TITLE
Question: how do I use the library to verify a commitment?

### ISSUE THREAD

**Issue by crudalex** [meilof/pysnark#10]
I notice there are PrivVal, ggh_hash, and the snark decorator available. 

But how to put them together to prove and verify a commitment?

**Comment by meilof** [comment]
I have added an example, hash.py, to demonstrate. It is used to prove knowledge of a preimage 1011101011101 of hash 3815100955245901773194254220410253439371965309251566846530536701111841134788. 

In this case, PrivVal is used to generate the witness, i.e., the preimage of the commitment. A PrivVal corresponds to a single numeric value of the witness, so it needs to be generated for each bit. Then, ggh_hash can be used to compute the hash. h.val() can be used to output the hash as a public output, or assert_equals can be used to demand that it is equal to a hash value that is given as a hardcoded value or a public input to the program.

In examples/hash_string.py, there is a more elaborate example, in which knowledge of a string preimage is proven.

These examples do not use a random seed in the hash but that could be added.

When making these examples I found out that there is a mistake in the python-libsnark library that pysnark uses. As a result, verification will fail for the examples above. I have fixed that mistake and will upload new binaries to pypi.

Hope that helps.
